clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -12397,6 +12397,9 @@ def warn_attr_min_eq_max:  Warning<
 def err_hlsl_attribute_number_arguments_insufficient_shader_model: Error<
   "attribute %0 with %1 arguments requires shader model %2 or greater">;
 
+def ext_hlsl_auto_type_specifier : ExtWarn<
+  "'auto' type specifier is a HLSL 202y extension">, InGroup<HLSL202y>;
```

Comment:
Maybe we can't do it because of the different warning groups, but it'd be nice if this could reuse `ext_auto_type_specifier` since it's basically the same warning. Is that possible?